rules/rulesets.py

```diff
@@ -1,49 +1,90 @@
-from .predicates import predicate
+from .predicates import predicate, Predicate
+from inspect import isfunction
 
 
 class RuleSet(dict):
+    def __init__(self, *args, **kwargs):
+        self.update(*args, **kwargs)
+
     def test_rule(self, name, *args, **kwargs):
+        # return name in self and self[name]['pred'].test(*args, **kwargs)
         return name in self and self[name].test(*args, **kwargs)
 
     def rule_exists(self, name):
         return name in self
 
-    def add_rule(self, name, pred):
+    def rule_verbose_name(self, name):
+        return self["get_%s_display" % name] or name
+
+    def add_rule(self, name, pred, verbose_name=None):
         if name in self:
             raise KeyError("A rule with name `%s` already exists" % name)
-        self[name] = pred
 
-    def set_rule(self, name, pred):
-        self[name] = pred
+        self[name] = {'pred': predicate(pred), 'verbose_name': verbose_name}
+
+    def set_rule(self, name, pred, verbose_name=None):
+        self[name] = {'pred': predicate(pred), 'verbose_name': verbose_name}
 
     def remove_rule(self, name):
         del self[name]
 
     def __setitem__(self, name, pred):
-        fn = predicate(pred)
-        super(RuleSet, self).__setitem__(name, fn)
-
+        if isfunction(pred):
+            # If a function as passed in (as might be done with legacy django-rules)
+            #   convert the value to the dictionary form
+            pred = {'pred': predicate(pred), 'verbose_name': verbose_name}
+
+        if isinstance(pred, dict):
+            # Check if internal pred is already a Predicate, or an basic
+            #   unwrapped function. Wrap as a Predicate if needed
+            if not isinstance(pred['pred'], Predicate):
+                pred['pred'] = predicate(pred['pred'])
+
+        super(RuleSet, self).__setitem__(name, pred)
+
+    def __getitem__(self, name):
+        def _check_name(_name):
+            if (not super(RuleSet, self).__contains__(_name)):
+                raise KeyError("Provided name '`%s`' not found" % _name)
+
+        if name[0] != '_':
+            prefix = "get_"
+            suffix = "_display"
+            if name.startswith(prefix) and name.endswith(suffix):
+                _name = name[len(prefix):-len(suffix)]
+                _check_name(_name)
+                return super(RuleSet, self).__getitem__(_name)['verbose_name']
+
+            _check_name(name)            
```

Comment:
All this can just be:

```suggestion
        prefix = "get_"
        suffix = "_display"
        if name.startswith(prefix) and name.endswith(suffix):
            _name = name[len(prefix):-len(suffix)]
            return super(RuleSet, self).__getitem__(_name)['verbose_name']
```

Right?